### DISCUSSION TITLE
Proposal for compressed font data? Any chance to get it accepted?

### DISCUSSION THREAD

**Discussion by holgerlembke** [moononournation/Arduino_GFX#330] [General]
Hi.

The other night I thought about a simple statistic-analysis+RLE-compression scheme for the fonts bitmap.

I just wrote a small test program to calc the ratio: 

Numbers are:
- FreeMono8pt7b 19% saved
- FreeMono60pt7b 47% saved
- FreeMonoOblique66pt7b 47% saved
- Clock font FreeMono12pt7b 81% saved
- Clock font FreeMono72pt7b 46% saved

The overall expectation would follow that pattern: small fonts at about 20%, larger fonts at the 45% ballpark figure.

Decompression is super simple, needs about 18 bytes ram and some code. End user change would be zero, "drawChar" will detect the new font format and just use a different methode. Again, code change will be minimal.

Process would be: take the current font file and process it into a new file.

Of course, the resulting fonts are incompatible to current tools like [Adafruit-GFX-Font-Customiser](https://github.com/tchapi/Adafruit-GFX-Font-Customiser). But the sources still exist, so simply reprocess them...

Any chance that there would be a wider acceptance of these fonts to be used?

**Comment by moononournation** [comment]
U8g font support compression, support Unicode and easy to generate custom font. That's why Arduino_GFX added support U8g font.
